src/components/attendee/VideoComms/types.tsx

```diff
@@ -0,0 +1,94 @@
+import Twilio from "twilio-video";
+
+export enum VideoCommsStatus {
+  Disconnected = "DISCONNECTED",
+  Connecting = "CONNECTING",
+  Connected = "CONNECTED",
+  Errored = "ERRORED",
+}
+
+export interface VideoCommsProviderProps {
+  userId: string;
+  children: React.ReactNode;
+}
+
+interface Track {
+  kind: "video" | "audio";
+  id: string;
+  enabled: boolean;
+}
+
+export enum VideoSource {
+  Webcam = "WEBCAM",
+  Screenshare = "SCREENSHARE",
+}
+
+export interface VideoTrack extends Track {
```

Comment:
Not sure if `interface`+`extends` would be better than just a union with `type`+`&`.
After all, why even have a base type?

Can just make them separate and add
`export type Track = VideoTrack | AudioTrack` 
with the `kind` already fixed at one specific value for them.

Reply:
Twilio has different types for Video and Audio with no shared base. This has led to some annoying code in our implementation. e.g. look at `Tracklike`. That wouldn't be needed if they had a shared base interface.

Can't say I care particularly strongly though.

Reply:
I always consider it a code smell naming something `*Base`. There is usually a better way of dealing with. In this case, I think union type is more robust approach as it doesn't try to invent shared interface but gives you the ability to use one or the other.